arithmetization/src/main/java/net/consensys/linea/zktracer/module/wcp/Wcp.java

```diff
@@ -21,28 +21,49 @@
 import static net.consensys.linea.zktracer.module.wcp.WcpOperation.ISZERObv;
 import static net.consensys.linea.zktracer.module.wcp.WcpOperation.LEQbv;
 import static net.consensys.linea.zktracer.module.wcp.WcpOperation.LTbv;
+import static net.consensys.linea.zktracer.module.wcp.WcpOperation.SGTbv;
+import static net.consensys.linea.zktracer.module.wcp.WcpOperation.SLTbv;
 
 import java.nio.MappedByteBuffer;
 import java.util.List;
 
-import lombok.Getter;
 import lombok.RequiredArgsConstructor;
-import lombok.experimental.Accessors;
 import net.consensys.linea.zktracer.ColumnHeader;
-import net.consensys.linea.zktracer.container.module.OperationSetModule;
+import net.consensys.linea.zktracer.container.module.Module;
 import net.consensys.linea.zktracer.container.stacked.CountOnlyOperation;
 import net.consensys.linea.zktracer.container.stacked.StackedSet;
 import net.consensys.linea.zktracer.opcode.OpCode;
 import org.apache.tuweni.bytes.Bytes;
 import org.apache.tuweni.bytes.Bytes32;
 import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.worldstate.WorldView;
 
 @RequiredArgsConstructor
-@Getter
-@Accessors(fluent = true)
-public class Wcp implements OperationSetModule<WcpOperation> {
-
-  private final StackedSet<WcpOperation> operations = new StackedSet<>();
+public class Wcp implements Module {
+
+  private final StackedSet<WcpOperation> ltOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> leqOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> gtOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> geqOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> sltOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> sgtOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> eqOperations = new StackedSet<>();
+  private final StackedSet<WcpOperation> isZeroOperations = new StackedSet<>();
+
+  /**
+   * For perf, we split the WcpOperations into different StackedSet in order to - have smaller
+   * set,thus faster to check for equality - remove the opcode value when checking the equality
+   */
+  private final List<StackedSet<WcpOperation>> operations =
```

Comment:
He already has a method which acts on a switch/case depending on the opcode so I thought it could be automated with a method but `# opcodes != # StackedSets` and there's no direct mapping unfortunately. So unless we can have a loop to automate this I think the extra field might be fine as it's an ArrayList.